extensions/wikia/FacebookClient/scripts/FacebookClient.XFBML.js

```diff
@@ -2,19 +2,34 @@
 
 /**
  * Render FB XML tags emebedded into wikitext
- * This file is loaded via ResourceLoader only when FB XML tags are present on the page
  */
 $(function () {
 	'use strict';
 
 	// load api on page load and on demand
 	mw.hook('wikipage.content').add(function ($content) {
-		$.loadFacebookAPI()
-			.done(function () {
-				$('.sso-login').removeClass('hidden');
 
-				// scan the new content for any fb tags
-				FB.XFBML.parse($content[0]);
-			});
+		if ((xfbmlTagsOnPage())) {
+			if (facebookSDKNotLoaded()) {
+				// XFBML tags rendered automatically when SDK loads
+				$.loadFacebookSDK();
+			} else {
+				renderXFBMLTags();
```

Comment:
No worries at all, I should have reached out to you when I noticed it rather than trying to work around it. I merged your fix it and it works great. And it does fix point number 1, so I'm taking that `fbxml: true` out of the config so we can just call `FB.XFBML.parse($content[0]);` manually whenever the wikipage.content event fires.